### PR TITLE
LB-1495: lb-radio endpoint is affected by letter case

### PR DESCRIPTION
**Pull request by ericd23**
# Problem
Basically, upper case tag won't do anything. For example: https://api.listenbrainz.org/1/lb-radio/tags?tag=Jazz&begin_percent=0&end_percent=50&count=50 gives 0 results. It's because PG's WHERE clause is case-sensitive.

# Solution
Convert all tags into lower case when loading the query params. Also add a integration test for this endpoint.

### CODE REVIEW

File: listenbrainz/webserver/views/api.py [hunk 1]
```diff
@@ -722,6 +722,7 @@ def get_tags_dataset():
     tag = request.args.getlist("tag")
     if tag is None:
         raise APIBadRequest("tag param is missing")
+    tag = [t.lower() for t in tag]
```
Comment (amCap1712) [inline listenbrainz/webserver/views/api.py:725]: I am not sure if all tag names in the database are lower cased. I'll check if that is the case, then the fix would work. If not, a case insensitive match should be performed in the SQL query. Will update.

Comment (MonkeyDo) [reply listenbrainz/webserver/views/api.py:725]: Ping @amCap1712

Comment (mayhem) [reply listenbrainz/webserver/views/api.py:725]: It appears that all tags are in fact lowercase:

```
musicbrainz_db=# select count(distinct(lower(name))) from tag;
 count  
--------
 213811
(1 row)

musicbrainz_db=# select count(*) from tag;
 count  
--------
 213811
(1 row)
```

**Comment by mayhem** [comment]
I feel that this isn't a problem that should be fixed in ListenBrainz, its a troi problem. I've fixed this in troi, so closing this PR. Thanks for submitting the PR, much appreciated!

**Pull request closed without merging**